code/script.js

```diff
@@ -262,82 +322,146 @@ const selectQuestion = () => {
+  if (keep)   {
+      alert ( `Super! the character has ${value} eyes! Let's keep all the characters with ${value} eyes.`)
+
+   } else {
+     alert(`Nope! The character does not have ${value} eyes removing all characters with ${value} eyes.`)
+      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+    }
+  } else if (category === 'look') {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
+      alert(`Booh yeah! the character looks ${attribute} Let's keep all the characters that look ${attribute}`)
+
     } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+      alert(`Nope! The character doesn't look ${attribute} Removing all the characters that look ${attribute}`)
+    } 
+
+  } else if (category === 'other') {
+    if (keep) {
+      alert(` Yes, the character is a ${attribute}. keeping all the characters that are ${attribute}`)
+    
+    } else {
+      alert(`Nope! The character is not a ${attribute} Removing all the characters that are ${attribute}s`)
     }
+
   }
 
   // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
-
+  if (keep) {
+    //right guess
+  charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
+  } else {
+    //wrong guess
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) 
+  }
+  generateBoard()
   // Invoke a function to redraw the board with the remaining people.
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (suspect) => {
   // store the interaction from the player in a variable.
+  const userConfirms = confirm(`Are you sure you want to guess on ${suspect} ?`)
+  if (userConfirms) {
+    checkMyGuess(suspect)
+  }
   // remember the confirm() ?
   // If the player wants to guess, invoke the checkMyGuess function.
 }
 
 // If you confirm, this function is invoked
 const checkMyGuess = (suspect) => {
   // 1. Check if the suspect is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
+  if (suspect === secret.name) {
+// 2. Set a Message to show in the win or lose section accordingly
+   winOrLoseText.innerHTML = `You thought it was ${suspect} and you were right!`
+  } else {
+    winOrLoseText.innerHTML = `Sorry, it's not ${suspect}. It was ${secret.name} Shall we give it another go?`
+  }
+  
   // 3. Show the win or lose section
+  winOrLose.style.display = 'flex'
   // 4. Hide the game board
+  board.style.display = 'none'
 }
 
 // Invokes the start function when website is loaded
-start()
+start();
```

Comment:
You could go without semi colon here, in order to keep the same semi colon approach all way through.